apps/client-ui/scripts/backend-discovery-service.sh

```
@@ -0,0 +1,523 @@
+    local cache_key="$1"
+    local data="$2"
+    local cache_file="${CACHE_DIR}/${cache_key}.json"
+    
+    echo "$data" > "$cache_file"
+    print_debug "Saved to cache: $cache_key"
+}
+
+# Function to clear cache
+clear_cache() {
+    if [[ -d "$CACHE_DIR" ]]; then
+        rm -rf "${CACHE_DIR}"/*
```

Shellcheck (0.10.0)

[warning] Use "${var:?}" to ensure this never expands to /* .

(SC2115)